snapshots/windows/windows.go

```diff
@@ -42,50 +75,298 @@ func NewSnapshotter(root string) (snapshots.Snapshotter, error) {
 //
 // Should be used for parent resolution, existence checks and to discern
 // the kind of snapshot.
-func (o *snapshotter) Stat(ctx context.Context, key string) (snapshots.Info, error) {
-	panic("not implemented")
+func (s *snapshotter) Stat(ctx context.Context, key string) (snapshots.Info, error) {
+	ctx, t, err := s.ms.TransactionContext(ctx, false)
+	if err != nil {
+		return snapshots.Info{}, err
+	}
+	defer t.Rollback()
+
+	_, info, _, err := storage.GetInfo(ctx, key)
+	if err != nil {
+		return snapshots.Info{}, err
+	}
+
+	return info, nil
 }
 
-func (o *snapshotter) Update(ctx context.Context, info snapshots.Info, fieldpaths ...string) (snapshots.Info, error) {
-	panic("not implemented")
+func (s *snapshotter) Update(ctx context.Context, info snapshots.Info, fieldpaths ...string) (snapshots.Info, error) {
+	ctx, t, err := s.ms.TransactionContext(ctx, true)
+	if err != nil {
+		return snapshots.Info{}, err
+	}
+
+	var committed bool
+	defer func() {
+		if committed == false {
+			rollbackWithLogging(ctx, t)
+		}
+	}()
```

Comment:
What's the purpose of tracking commited?

Looking at [the implementation of `Tx.Rollback`](https://github.com/boltdb/bolt/blob/master/tx.go#L240-L247) I don't think this error is ever relevant.  The only error is "transaction was already closed", which I think would only occur if `commited == true`, so this will never log anything.

Maybe remove the `committed`, and ignore the error from `Rollback()` ? Then you can always just `defer tx.Rollback()`, which seems to be what other implementations are doing.

Reply:
Makes sense. For some reason I though that rollback shouldn't be called after a commit, but I don't know why. Changing this to be similar to other implementations

Reply:
Actually, the other snapshotters are doing almost the exact same thing such as https://github.com/containerd/containerd/blob/master/snapshots/overlay/overlay.go#L290 (`bool rollback` instead of `bool committed`, but the exact same idea) and https://github.com/containerd/containerd/blob/master/snapshots/overlay/overlay.go#L204 (setting `t = nil` to abandon the rollback instead of using a bool).

Same thing in naive. Looks like the other snapshotters should also be fixed to just use `defer tx.Rollback()` regardless to me.